core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1305,66 +1305,102 @@ class KafkaApis(val requestChannel: RequestChannel,
   }
 
   def handleFindCoordinatorRequest(request: RequestChannel.Request): Unit = {
+    val version = request.header.apiVersion
+    if (version < 4) {
+      handleFindCoordinatorRequestLessThanV4(request)
+    } else {
+      handleFindCoordinatorRequestV4AndAbove(request)
```

Comment:
Do we really need these separate methods? I would have thought we could just wrap the `findCoordinatorRequest.data.key` in a singleton List for the `version < 4` and otherwise they'd be handled the same. If there is a reason I think it deserves a comment.

Reply:
We need a different `createResponse()` for each version. Both methods call `getCoordinator()` and have no logic.

Reply:
Yes, obvious now you point it out! I guess I can't see a better way of abstracting over that.